src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -137,7 +138,10 @@ void VehicleInfoPendingResumptionHandler::RaiseFinishedPendingResumption(
     SDL_LOG_DEBUG("Subscribe " << app->app_id() << "  to " << subscription);
     ext.subscribeToVehicleInfo(subscription);
   }
-
+  application_manager_.TerminateRequest(
```

Comment:
@ychernysheva I believe you can just use `unsubscribe_from_event()` so handler will ignore any events caused by that request. In that case you don't need `corr_id_`

Reply:
@AKalinich-Luxoft Yes, it seems that proposed approach also works well, but I'm still confused about it. I've checked SDL logs after applying of this change and I see that in this case this request also will be retained in RequestController, when corresponding app is already unregistered. After this we also will have OnTimeOut() call for this request, but in this case it probably doesn't cause any problems (but I cannot guarantee it now). 
If you think that this is okay, I'll  apply this changes

Reply:
@ychernysheva I prefer to use subscription-based approach here due to several reasons:
1. Because the same component is using `subscribe_to_event()` to make a subscription so it make sense to do an opposite action when response is not expected anymore
2. I consider `TeminateRequest` like a tool for a manual control of requests lifetime. Such requests should also have to call `SetAllowedToTerminate(false)` if they are expected to be teminated like that. In your case, requests have an automatic lifetime control - they are destroyed automatically after `SendResponse`. That's why I don't think this is a good idea just to kill it with `TerminateRequest`
3. Handler is the only component which creates and subscribes to that request. Subscription is made by `correlation_id` which is unique, that's why if handler needs to unsubscribe and then subscribe to the next SubsribeVD request there will be no conflict and any side effects as it will be another isolated request with a separate `correlation_id`

Reply:
@AKalinich-Luxoft Agree, fix changed in bb89eef7559c058ac5fa835f40e42c5d150aedc8